### PR TITLE
REZ IT -> WEAR IT

### PR DESCRIPTION
**Pull request by zfox23**
**NOTE:** This code BREAKS Commerce Checkout and Purchases until [this data-web PR](https://github.com/highfidelity/data-web/pull/1073) is merged from Staging into Stable!!! DO NOT TEST until I put the `QA Ready` label on this PR.

# Test Plan

## Prerequisites

Currently, you must manually enable "commerce" by adding `"commerce": true,` in your Interface.json (with Interface not running), or by opening Interface Javascript console and evaluating `Settings.setValue("commerce", true)` and restarting.

If your wallet is already set up OR if you have previously set up a wallet on a different build (and thus you have a record of your wallet on the backend), you must perform the following steps to rest your wallet:
1. Open the Wallet app. If you're just resetting your wallet because there's a record of a previous wallet on the backend, go through setup.
2. Go to the "Help" tab in the Wallet app
3. Press the red "DBG: RST Wallet" button at the top of the window

If your wallet is set up but broken (e.g., from testing something that didn't work) such that you are prompted for a passphrase that is never accepted, you need to:
1. Quit interface and delete `%appdata%/Interface <build #>/<username.hifikey>`
2. Restart Interface, set up wallet, and then press the red `DBG: RST WALLET` button as above. Now you are in a truly clean state for wallet setup.

## Tests
1. Set up your Wallet and go to your Sandbox. On the Domain Settings Page, ensure that `localhost` has `Certified (Temp) Rez` permissions.
2. Buy a `Test Beach Ball` from the Marketplace. On the Purchase Confirmation page, verify that the red button says "REZ IT"
3. Buy a `Purple Scarf` from the Marketplace. On the Purchase Confirmation page, verify that the red button says "WEAR IT"
4. On your Purchases screen, verify that the red button next to the Purple Scarf says "WEAR IT"
5. On your Purchases screen, verify that the red button next to the Test Beach Ball says "REZ IT"
6. Close the Purchases screen
7. On your Domain Settings Page, make it so `localhost` does NOT have `Certified (Temp) Rez` permissions.
8. Buy a `Test Beach Ball` from the Marketplace. On the Purchase Confirmation page, verify that the "REZ IT" button is disabled and that there is red text below the "REZ IT" button explaining that you don't have permission to rez.
9. Buy a `Purple Scarf` from the Marketplace. On the Purchase Confirmation page, verify that the red button says "WEAR IT" and that it is enabled
10. On your Purchases screen, verify that the red button next to both Purple Scarves says "WEAR IT" and is enabled
11. On your Purchases screen, verify that the button next to the Test Beach Ball says "REZ IT" and is disabled
12. On your Purchases screen, verify that you see a red box at the top of the screen that explains that you don't have Rez permissions.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4265/HighFidelity-Beta-PR11629-322bac035ce08f7756ddf7eb12be013045a9ea86.dmg

**Comment by zfox23** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4266/HighFidelity-Beta-PR11629-322bac035ce08f7756ddf7eb12be013045a9ea86.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4266/HighFidelity-Beta-PR11629-322bac035ce08f7756ddf7eb12be013045a9ea86.exe

File: interface/resources/qml/hifi/commerce/checkout/Checkout.qml [hunk 1]
```diff
@@ -39,7 +39,8 @@ Rectangle {
     property bool itemIsJson: true;
     property bool shouldBuyWithControlledFailure: false;
     property bool debugCheckoutSuccess: false;
-    property bool canRezCertifiedItems: Entities.canRezCertified || Entities.canRezTmpCertified;
+    property bool canRezCertifiedItems: Entities.canRezCertified() || Entities.canRezTmpCertified();
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/commerce/checkout/Checkout.qml]: :-)

File: scripts/system/marketplaces/marketplaces.js [hunk 2]
```diff
@@ -74,8 +74,8 @@
                     itemName: 'Test Flaregun',
                     itemPrice: (debugError ? 10 : 17),
                     itemHref: 'http://mpassets.highfidelity.com/0d90d21c-ce7a-4990-ad18-e9d2cf991027-v1/flaregun.json',
-                },
-                canRezCertifiedItems: Entities.canRezCertified || Entities.canRezTmpCertified
+                    categories: ["Wearables", "Miscellaneous"]
```
Comment (howard-stearns) [inline scripts/system/marketplaces/marketplaces.js]: ? I don't understand this part.

File: interface/resources/qml/hifi/commerce/checkout/Checkout.qml (hunk 1 above)
Comment (zfox23) [reply interface/resources/qml/hifi/commerce/checkout/Checkout.qml]: Hehe :)

File: scripts/system/marketplaces/marketplaces.js (hunk 2 above)
Comment (zfox23) [reply scripts/system/marketplaces/marketplaces.js]: Yeah - it's unclear without just _a couple_ of extra lines of code being visible above.

This is for when the local JS variable `debugCheckout` is `true` (it's `false`) by default. When that's true, clicking the MARKET button on the HUD goes straight to checking out a `test flaregun`, and the test uses these two categories.

This code has no effect on production; I was just using it to test (the test passed).

Comment (howard-stearns) [reply scripts/system/marketplaces/marketplaces.js]: got it

**Comment by zfox23** [comment]
@howard-stearns is testing.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4289/HighFidelity-Beta-PR11629-b3acdfa9d85fb11065b87023a313694459900ed2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4289/HighFidelity-Beta-PR11629-b3acdfa9d85fb11065b87023a313694459900ed2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4311/HighFidelity-Beta-PR11629-b3acdfa9d85fb11065b87023a313694459900ed2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4311/HighFidelity-Beta-PR11629-b3acdfa9d85fb11065b87023a313694459900ed2.exe

**Pull request merged**